core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.controller.ts

```diff
@@ -59,34 +62,46 @@ angular.module('oppia').directive('topicsAndSkillsDashboardPage', [
         '$http', '$rootScope', '$scope', '$uibModal', '$window',
         'AlertsService', 'RubricObjectFactory', 'SkillCreationService',
         'SkillObjectFactory', 'TopicCreationService',
-        'TopicsAndSkillsDashboardBackendApiService', 'UrlInterpolationService',
+        'TopicsAndSkillsDashboardBackendApiService',
+        'TopicsAndSkillsDashboardPageService', 'UrlInterpolationService',
         'EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED',
         'EVENT_TYPE_SKILL_CREATION_ENABLED',
         'EVENT_TYPE_TOPIC_CREATION_ENABLED',
         'FATAL_ERROR_CODES', 'SKILL_DIFFICULTIES',
         'MAX_CHARS_IN_SKILL_DESCRIPTION', 'SKILL_DESCRIPTION_STATUS_VALUES',
+        'TOPIC_CATEGORIES', 'E_SORT_OPTIONS', 'E_PUBLISHED_OPTIONS',
         function(
             $http, $rootScope, $scope, $uibModal, $window,
             AlertsService, RubricObjectFactory, SkillCreationService,
             SkillObjectFactory, TopicCreationService,
-            TopicsAndSkillsDashboardBackendApiService, UrlInterpolationService,
+            TopicsAndSkillsDashboardBackendApiService,
+            TopicsAndSkillsDashboardPageService, UrlInterpolationService,
             EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED,
             EVENT_TYPE_SKILL_CREATION_ENABLED,
             EVENT_TYPE_TOPIC_CREATION_ENABLED,
             FATAL_ERROR_CODES, SKILL_DIFFICULTIES,
-            MAX_CHARS_IN_SKILL_DESCRIPTION, SKILL_DESCRIPTION_STATUS_VALUES) {
+            MAX_CHARS_IN_SKILL_DESCRIPTION, SKILL_DESCRIPTION_STATUS_VALUES,
+            TOPIC_CATEGORIES, E_SORT_OPTIONS, E_PUBLISHED_OPTIONS) {
           var ctrl = this;
-          var _initDashboard = function(stayInSameTab) {
+          ctrl._initDashboard = function(stayInSameTab) {
             TopicsAndSkillsDashboardBackendApiService.fetchDashboardData().then(
               function(response) {
+                ctrl.totalTopicSummaries = response.data.topic_summary_dicts;
```

Comment:
Try and convert the response into a domain object or separate args within the backend API service, and only then pass it to the controller layer.

Reply:
Done.